packages/mgt-components/src/components/mgt-todo/mgt-todo.ts

```diff
@@ -700,17 +703,25 @@ export class MgtTodo extends MgtTasksBase {
     } else {
       this._newTaskDueDate = null;
     }
+  };
 
+  private readonly handleDateUpdate = (e: Event) => {
     const task = this._taskBeingUpdated;
+    if (task) {
+      const value = (e.target as HTMLInputElement).value;
+      if (value) {
+        this._newTaskDueDate = new Date(value + 'T17:00');
+      } else {
+        this._newTaskDueDate = null;
+      }
 
-    if (task?.dueDateTime && this._newTaskDueDate) {
-      this._isChangedDueDate = new Date(task.dueDateTime.dateTime) !== this._newTaskDueDate;
-    } else if ((task?.dueDateTime && !this._newTaskDueDate) || (!task?.dueDateTime && this._newTaskDueDate)) {
-      this._isChangedDueDate = true;
-    } else {
-      this._isChangedDueDate = false;
+      if (task.dueDateTime && this._newTaskDueDate) {
+        this._isChangedDueDate = new Date(task.dueDateTime.dateTime) !== this._newTaskDueDate;
+      } else if ((task.dueDateTime && !this._newTaskDueDate) || (!task.dueDateTime && this._newTaskDueDate)) {
```

Comment:
given that the case where both are truthy is covered as the first condition isn't this line a simple or?
```suggestion
      } else if (task.dueDateTime || this._newTaskDueDate) {
```